### PR TITLE
Add documentation on caching

### PR DESCRIPTION
**Pull request by richaagarwal**

### CODE REVIEW

File: docs/caching.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Caching
+
+### Akamai
+
+We use Akamai, a content delivery network, (https://www.akamai.com/), to cache the entirety of consumerfinance.gov (but not our development servers). We invalidate any given page in Wagtail when it is published or unpublished (by hooking up the custom class `AkamaiBackend` (in https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/v1/models/akamai_backend.py) to Wagtail's frontend cache invalidator (http://docs.wagtail.io/en/v1.9/reference/contrib/frontendcache.html). By default, we clear the Akamai cache any time we deploy.
```
Comment (anselmbradford) [inline docs/caching.md]: TRIVIAL: To shorten the text we could hide the URLs where applicable. E.g. Have Akamai text as a link with `[Akamai](https://www.akamai.com/)`

Comment (anselmbradford) [reply docs/caching.md]: Does Akamai cover all subdomains of cf.gov? E.g. is beta.consumerfinance.gov behind Akamai? If so we possibly could say `*.consumerfinance.gov`. If not, we could say `www.consumerfinance.gov`.

File: docs/caching.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+There are certain pages that do not live in Wagtail or are impacted by changes on another page (imagine our newsroom page that lists titles of other pages - https://www.consumerfinance.gov/about-us/newsroom/) or another process (imagine data from Socrata gets updated) and thus will display outdated content until the page's time to live (TTL) has expired, a deploy has happened, or if someone manually invalidates that page. Our default TTL is 24 hours.
+
+### Django caching
+
+Starting in December 2017, we use template fragment caching (https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/v1/fragment_cache_extension.py) to cache all or part of a template.  It is enabled on our "post previews", snippets of a page that we display on results of filterable pages (e.g. http://consumerfinance.gov/about-us/blog, https://www.consumerfinance.gov/data-research/research-reports/). 
+
+It can easily be enabled on other templates. See https://github.com/cfpb/cfgov-refresh/pull/3663/files as an example of the code that would need to be introduced to cache a new fragment.
```
Comment (anselmbradford) [inline docs/caching.md]: TRIVIAL: Similar to above we could hide the actual URL behind a link to shorten the text:
```
See [this PR](https://github.com/cfpb/cfgov-refresh/pull/3663/files) as an example of the code…
```

File: docs/caching.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+When a page gets published, it will update the post preview cache for that particular page.  However, if there are code changes that impact the page's content, or the post preview template itself gets updated, the entire post preview cache will need to be manually cleared. Clearing this particular cache could be an option when deploying, as it is with Akamai, but should not be a default since most deploys wouldn't impact the code in question.  Currently, the manual way to do this would be to run the following from a production server:
+
+```
+from django.core.cache import caches
+
+caches['post_preview'].clear()
+```
+
+To run the application locally with caching for post previews enabled, run `ENABLE_POST_PREVIEW_CACHE=1 ./runserver.sh`
```
Comment (anselmbradford) [inline docs/caching.md]: Should `ENABLE_POST_PREVIEW_CACHE` be in `.env_SAMPLE`?

File: docs/caching.md (hunk 2 above)
Comment (richaagarwal) [reply docs/caching.md]: Hmmm yeah I could do that, but what's the value of obscuring the URL?

**Comment by richaagarwal** [comment]
@anselmbradford Thanks for the feedback, Ans!

**Pull request merged**